oak_remote_attestation/src/dice.rs

```diff
@@ -0,0 +1,185 @@
+//
+// Copyright 2023 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use crate::proto::oak::attestation::v1::{
+    CertificateAuthority, DiceData, Evidence, LayerEvidence, RootLayerEvidence, TeePlatform,
+};
+use alloc::vec::Vec;
+use anyhow::{anyhow, Context};
+use ciborium::{from_reader, into_writer, Value};
+use coset::{
+    cwt::{ClaimName, ClaimsSet},
+    CborSerializable,
+};
+use oak_dice::{
+    cert::{generate_eca_certificate, get_claims_set_from_certificate_bytes},
+    evidence::Stage0DiceData,
+};
+use p256::ecdsa::SigningKey;
+
+/// The actual size used when encoding a Nist P256 private key.
+const P256_PRIVATE_KEY_SIZE: usize = 32;
+
+/// The size of the attestation report generated by AMD SEV-SNP.
+///
+/// See Table 21 in <https://www.amd.com/system/files/TechDocs/56860.pdf>.
+const AMD_SEV_SNP_ATTESTATION_REPORT_SIZE: usize = 1184;
+
+/// Builds the DICE evidence and certificate authority for the next DICE layer.
+pub struct DiceDataBuilder {
```

Comment:
Should it be `DiceBuilder`? Since it's not only building the DICE data but adds an Evidence in the chain as well.

Reply:
Done.